pkg/cmd/update.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/daytonaio/daytona/internal"
+	"github.com/inconshreveable/go-update"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/semver"
+	"net/http"
+	"net/url"
+	"regexp"
+	"runtime"
+	"strings"
+)
+
+const (
+	githubReleaseURL = "https://api.github.com/repos/daytonaio/daytona/releases"
+	baseDownloadURL  = "https://download.daytona.io/daytona/"
+)
+
+type GitHubRelease struct {
+	TagName   string `json:"tag_name"`
+	ChangeLog string `json:"body"`
+}
+
+var versionFlag string
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Update Daytona CLI",
+	Long:  "Update Daytona CLI to the latest version",
+	Args:  cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var version string
+		var changeLog string
+		currentVersion := internal.Version
+		if versionFlag == "" {
+			release, err := fetchLatestRelase()
+			if err != nil {
+				return err
+			}
+			version = release.TagName
+			changeLog = release.ChangeLog
+		} else {
+			semverRegex := `^v?(\d+\.\d+\.\d+)$`
+			matched, err := regexp.MatchString(semverRegex, versionFlag)
+			if err != nil {
+				return err
+			}
+			if !matched {
+				return errors.New("invalid version format: expected 'vX.Y.X' or 'X.Y.Z'")
+			}
+			if !strings.HasPrefix(versionFlag, "v") {
+				versionFlag = "v" + versionFlag
+			}
+			release, err := fetchVersionRelease(versionFlag)
+			if err != nil {
+				return err
+			}
+			version = release.TagName
+			changeLog = release.ChangeLog
+		}
+		if semver.Compare(version, currentVersion) <= 0 {
+			fmt.Println("Current version is greater than the version you are trying to update to")
+			return nil
+		}
+		if semver.Compare(currentVersion, version) != 0 {
+			changeLog += "\nThere are more changes in the version. Please visit https://github.com/daytonaio/daytona/releases for all the complete chnageLog"
```

Comment:
Change it to: `changelog += "\n\nThere might be more important changes since you updated. Please visit https://github.com/daytonaio/daytona/releases for the complete changelog\n"`

Also, seems to me that you removed the functionality to check if the current version is behind the updated one by more than one release.
This line should only be shown then.
In the case in the screenshot, this line should not be rendered.

![Screenshot 2024-12-10 at 17 18 02](https://github.com/user-attachments/assets/7d3be7a1-0658-4c65-8e39-597ff10d0295)

Reply:
i got mistaken and removed it and added the semantic compare, now i have currently added the functionality.

Reply:
![Screenshot 2024-12-11 at 12 23 18](https://github.com/user-attachments/assets/37b0c99a-8efd-4c8b-940a-56e1edf691a2)

@Abiji-2020 now I can't even update if I'm not more than 1 version behind.